### PR TITLE
Added TTL to settings.

### LINKED ISSUE #128: Allow adjusting TTL for messages
currently default TTL is 24 hours for regular messages. We should allow this to be edited to a certain extent.

### PR DESCRIPTION
**Pull request by Mikunj**
Resolves oxen-io/session-desktop-temp#1210.

The current minimum is 12 hours (1/2 day) and the max is 96 hours (4 days).

screen shot 2019-01-14 at 12 41 40 pm

### CODE REVIEW

File: js/background.js [hunk 1]
```diff
@@ -237,6 +241,19 @@
         window.setMenuBarVisibility(!value);
       },
 
+      getMessageTTL: () => {
+        // Make sure the ttl is between a given range
+        const current = storage.get('message-ttl', 24);
+        return Math.max(MIN_MESSAGE_TTL, Math.min(current, MAX_MESSAGE_TTL));
+      },
+      setMessageTTL: value => {
```
[the remark this answers is not in the document]
Comment (Mikunj) [reply js/background.js]: Users can just edit the html to get over the limits.

File: js/background.js [hunk 2]
```diff
@@ -221,6 +221,10 @@
     }
     first = false;
 
+    // The min and max message ttl value
+    const MIN_MESSAGE_TTL = 12;
```
[the remark this answers is not in the document]
Comment (Mikunj) [reply js/background.js]: Just a temp limit, will need to speak with Kee to work it out.

File: settings.html [hunk 3]
```diff
@@ -112,6 +112,42 @@ <h3>{{ permissions }}</h3>
...
     <hr>
+    <div class='message-ttl-setting'>
+      <h3>{{ messageTTL }}</h3>
+      <div>{{ messageTTLSettingDescription }}</div>
+      <div id="warning">{{ messageTTLSettingWarning }}</div>
+      <div class='inputs'>
+        <input
+          name='message-ttl-setting'
+          id='message-ttl-setting'
+          type="range"
```
Comment (sachaaaaa) [inline settings.html]: nits: stick to using single quotes :)

Comment (Mikunj) [reply settings.html]: Woops, i'll fix it right up

**Review by BeaudanBrown — APPROVED** [review]
Seems good

**Comment by BeaudanBrown** [comment]
Should note that we will want to update the storage server to restrict the range of TTL that it accepts

**Pull request merged**